app/controllers/device_controller.rb

```diff
@@ -1,19 +1,23 @@
 class DeviceController < ApplicationController
   before_action :authorize_request, :only=>[:add_tag, :delete_tag]
 
-  def show
+  def index
     @devices = Device.all
   end
 
+  def show
+    @device = Device.find_by(display_name: params[:device])
+  end
+
   def raw_data
-    devices = Device.left_joins(:user)
+    devices = Device.left_joins(:user) # This join is still necessary for the future leaderboard to include usernames
     .select('devices.*, users.username, max / (cpus * cores_per_cpu) AS max_per_core')
     .order(:max_per_core)
     response = {}.tap do |res|
       max_device = devices.last
       res[:max_main] = max_device&.max_per_core&.round(3);
       res[:header] = {}.tap do |h|
-        h[:user] = 'User'
+        h[:user] = 'Name'
```

Comment:
I think for the time being we've got name-editing as something to add only if we have time for it - given that we're generating ALL the names, having some of them trail off with ellipsis seems odd. I think increasing the width is a good idea, and we can always change the colours/animals/adjectives to lower the maximum length that a name can be - for example swapping out `chicken` and `dragon` for shorter animals gives us two more characters because those two are the only animals longer than 5 characters.

The current naming logic is pretty simple, and it looks like the longest possible name would be 20 characters e.g. `QuickYellowChicken32`. The chosen words for the username generation were completely arbitrary so we can easily replace any of them with shorter words.

Reply:
Sure that makes sense :)

I was going to say users can increase their browser font size which would mess with the length but it seems that our font size is hard coded in atm so nvm :P

Reply:
Another issue I'm vaguely conscious of with the naming function is that it works on a loop:

1. Generate a name
2. If that name is taken, repeat step 1.

which works fine if a lot of the names are free, but if, say, there were only 10 names left to claim, the name generation would take a long time because it would keep rolling until it happens to generate an unclaimed name. Roughly 9000 generations to be exact. Granted, there are 100,000 possible names, but it's a consideration that the name-generating algorithm perfoms exponentially worse when we approach the limit on names.

Reply:
Good point. Partially thought out idea for future - could get all the permutations of 3 indices (0-9) and a number (0-99), randomise the order of the permutations using a known seed, then take the next permutation in the list for each new device.
 
E.g. Device number 3 gets the third permutation in the list: [2, 8, 4, 56]
So that device gets the name `#{ colour[2] }#{ adjective[8] }#{ animal[4] }#{ 56 }`